backend/remote-state/pg/backend.go

```diff
@@ -0,0 +1,88 @@
+package pg
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/helper/schema"
+	_ "github.com/lib/pq"
+)
+
+const (
+	statesTableName = "states"
+	statesIndexName = "states_by_name"
+)
+
+// New creates a new backend for Postgres remote state.
+func New() backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"conn_str": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Postgres connection string; a `postgres://` URL",
+			},
+
+			"schema_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Name of the automatically managed Postgres schema to store state",
+				Default:     "terraform_remote_backend",
```

Comment:
let's make this `terraform_remote_state`, since "remote backend" can have different meanings

Reply:
Done